src/ucl_util.c

```diff
@@ -40,9 +40,56 @@
 #endif
 
 #ifdef HAVE_LIBGEN_H
+#ifndef _MSC_VER
 #include <libgen.h> /* For dirname */
+#else
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+char *dirname(char *path)
+{
+    char drive[3];
+    char dir[512]; // TODO : use appropriate constants macro
+    char val[512]; // TODO : use appropriate constants macro
```

Comment:
Why not use `PATH_MAX` for dirname? For filename there is a dedicated constant in Unix.